src/clusterfuzz/_internal/bot/fuzzers/afl/launcher.py

```diff
@@ -1107,10 +1181,177 @@ def __init__(self,
+      A new_process.ProcessResult.
+    """
+
+    assert not testcase_path.isdigit(), ('We don\'t want to specify number of'
+                                         ' executions by accident.')
+    self.afl_setup()
+    self._executable_path = android.adb.get_adb_path()
+    self._default_args = ["shell"] + \
+                         self.get_afl_environment_variables() + \
+                         [android.util.get_device_path(self.target_path)]
+
+    device_target_path = android.util.get_device_path(testcase_path)
+    android.adb.copy_local_file_to_remote(testcase_path, device_target_path)
+
+    result = self.run_and_wait(additional_args=[device_target_path])
+
+    # Copy error to local.
+    android.adb.copy_remote_file_to_local(
+        android.util.get_device_path(self.stderr_file_path),
+        self.stderr_file_path)
+
+    self.verify_return_code(result, [134])
+    return result
+
+  def get_file_features(self, input_file_path, showmap_args):
+    """Get the features (edge hit counts) of |input_file_path| using
+    afl-showmap."""
+    # TODO(metzman): Figure out if we should worry about CPU affinity errors
+    # here.
+
+    filename = os.path.basename(input_file_path)
+    intput_file_showmap_results_file = os.path.join(self._showmap_results_dir, filename)
+    showmap_output = engine_common.read_data_from_file(intput_file_showmap_results_file)
+    return self.get_feature_tuple(showmap_output), False
+
+  def fuzz(self):
+    self._executable_path = android.adb.get_adb_path()
+    self._default_args = ["shell"]
+
+    self.initial_max_total_time = (
+        get_fuzz_timeout(
+            self.strategies.is_mutations_run, full_timeout=self.timeout) -
+        self.AFL_CLEAN_EXIT_TIME - self.SIGTERM_WAIT_TIME)
+
+    on_device_fuzzing_timeout = self.timeout - int(self.AFL_CLEAN_EXIT_TIME
+                                        + self.SIGTERM_WAIT_TIME
+                                        + self.DEVICE_FUZZING_CLEAN_EXIT_TIME)
+
+    self._copy_local_directories_to_device(
+        [
+            self.afl_input.input_directory,
+            os.path.dirname(os.path.realpath(self.target_path))
+        ])
+
+    device_output_dir = android.util.get_device_path(
+        self.afl_output.output_directory)
+
+    android.adb.create_directory_if_needed(device_output_dir)
+
+    self._fuzz_args = self.generate_afl_args(
+        afl_input=android.util.get_device_path(self.afl_input.input_directory),
+        afl_output=device_output_dir,
+        additional_args=[constants.FUZZING_TIMEOUT_FLAG + str(on_device_fuzzing_timeout)],
+        target_path=android.util.get_device_path(self.target_path))
+
+    fuzz_result = self.run_afl_fuzz(self._fuzz_args)
+
+    # Generate file features from fuzzing results.
+    device_script_path = os.path.join(
+        android.constants.DEVICE_FUZZING_DIR, "run.sh")
+
+    local_script_path = environment.get_root_directory() + \
+                        "/src/clusterfuzz/_internal/" \
```

Comment:
No don't do this. Use `__file__` to find the script instead of using an absolute path (which will break when developing).